md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -113,16 +120,16 @@
             Process parentProcessId = new Process();
             parentProcessId.setProcessId(processId);
             List<Process> listOfSubProcesses = new ArrayList<Process>();
-            Criteria listOfSubProcessCriteria = session.createCriteria(Process.class).add(Restrictions.eq("process", parentProcessId)).add(Restrictions.eq("deleteFlag", false));
+            Criteria listOfSubProcessCriteria = session.createCriteria(Process.class).add(Restrictions.eq(PROCESS, parentProcessId)).add(Restrictions.eq(DELETEFLAG, false));
             listOfSubProcesses = listOfSubProcessCriteria.list();
 
             ExecStatus runningExecState = new ExecStatus();
             runningExecState.setExecStateId(2);
-            Criteria stepsExecutionCheckCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq("execStatus", runningExecState));
-            if (listOfSubProcesses.size() > 0) {
+            Criteria stepsExecutionCheckCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq(EXECSTATUS, runningExecState));
+            if (!listOfSubProcesses.isEmpty()) {
                 for (Process stepsProcess : listOfSubProcesses) {
-                    stepsExecutionCheckCriteria.add(Restrictions.eq("process", stepsProcess));
-                    if (stepsExecutionCheckCriteria.list().size() > 0) {
+                    stepsExecutionCheckCriteria.add(Restrictions.eq(PROCESS, stepsProcess));
+                    if (!stepsExecutionCheckCriteria.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)